src/main/java/de/novatec/ddd/domainprimitives/validation/Constraints.java

```diff
@@ -201,6 +206,38 @@ private static boolean isLessThanOrEqual(Double value, Double maxSize) {
         );
     }
 
+    // Date
+
+    public static Consumer<Validation<Date>> isInFuture() {
+        return val -> val.constraint(
+                isNotNull(val.value()) && new Date().before(val.value()),
+                () -> format(ERROR_MESSAGE_IN_FUTURE_TEMPLATE, getValueFormatted(val))
+        );
+    }
+
+    public static Consumer<Validation<Date>> isInPast() {
+        return val -> val.constraint(
+                isNotNull(val.value()) && new Date().after(val.value()),
+                () -> format(ERROR_MESSAGE_IN_PAST_TEMPLATE, getValueFormatted(val))
+        );
+    }
+
+    // LocalDate
+
+    public static Consumer<Validation<LocalDate>> isInFutureLocalDate() {
```

Comment:
I would suggest to base this on the "Temporal" Interface and check based on implementation to offer wider support of java.time